### DISCUSSION TITLE
translation error (italian)

### DISCUSSION THREAD

**Discussion by painbrain81** [NightscoutFoundation/xDrip#2691] [General]
Hi everyone. In the Italian app, the "BG falling fast" and "BG raising fast" notifications are not translated. Can I then report this error to get it fixed?

thank you!

**Comment by Navid200** [comment]
I am curious about this.
Looking at speak readings settings, for speak alerts, the summary reads "Also give some spoken word messages".
That sounds like there is no guarantee.

I just enabled speak readings on my phone and speak alerts and forced it to trigger a high alert.  What it said was "High 6.5".  
It is possible that some alerts have been set up and some haven't.  I don't think this is a language problem.
Rather, I think spoken alerts does not say anything for rising fast in any language.  But, I cannot emphasize enough that I am not 100% sure.

**Comment by painbrain81** [comment]
No speaks alert but the write notification.
"bg falling fast" in the notification area (when bg falling down and the alert is enable of course) is "bg discesa rapida" in italian. 
![Screenshot_20230222_062751](https://user-images.githubusercontent.com/24631990/220531076-62c2f76f-1909-4e25-95b8-a4b416b29386.jpg)

**Reply by Navid200** [reply]
I see you are talking about the written notification.
Would you please post a screenshot when you can?

**Reply by Navid200** [reply]
@psonnera
Are all alert notifications translated?   For example, is rising fast notification translated?

**Reply by painbrain81** [reply]
All notifications are traslated. But BG falling and raising fast no :) (in italian, i don't know the other langages)

**Reply by psonnera** [reply]
@Navid200 I don't know, my phone is in English. I would have to check if all notifications are in the translation tables and not hard coded. AFAIK Italian is 100% but I'll check that.

**Reply by Navid200** [reply]
None of the logs are translated.  And they are not supposed to be translated.

Notifications should be translated.
Alerts in xDrip are based on notifications.  So, an alert that uses sound does that while a notification is shown.  I am not 100% sure all alert notifications are meant to be translated.

I will try to find out in the code if they are.

**Reply by painbrain81** [reply]
Infact it's not a log but a notification. No worry, i will screenshot next alarm so i can show you to help to fix :)

**Reply by Navid200** [reply]
I never said it was a log.

My point was that not everything in xDrip is translated.  Not everything is meant to be translated.
Sometimes it helps to have a conversation in order to figure out the root cause of what a user experiences.  Not every single post in a discussion is meant to be an answer to the original question.

**Reply by Navid200** [reply]
I asked someone else to join the conversation and he did.  I was responding to his post.

**Reply by painbrain81** [reply]
Sorry i dont understood right in this case :)

**Comment by painbrain81** [comment]
Here is an example. My app is in italian but not the notification. 
![Screenshot_20230222_185707_com teslacoilsw launcher](https://user-images.githubusercontent.com/24631990/220717022-a1a092b3-5e7e-4da9-a0d4-6c032b4bcf91.jpg)

**Reply by Navid200** [reply]
Thanks for the screenshot.

This is where that notification is coming from in the code:
https://github.com/NightscoutFoundation/xDrip/blob/fcd8063a4149c59d01bc668e191eaa3c6cee5c3e/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java#L916

It's not a string.  Therefore, it cannot be translated.

I'm not saying you are wrong.  I am saying this notification is not translated to any language currently.

**Reply by painbrain81** [reply]
ok thank you :)
